versal2.mk

```diff
@@ -0,0 +1,173 @@
+################################################################################
+# Following variables defines how the NS_USER (Non Secure User - Client
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+override COMPILE_NS_USER	:= 64
+override COMPILE_NS_KERNEL	:= 64
+override COMPILE_S_USER		:= 64
+override COMPILE_S_KERNEL	:= 64
+
+# Network support related packages:
+BR2_PACKAGE_DHCPCD	?= y
+BR2_PACKAGE_ETHTOOL	?= y
+BR2_PACKAGE_XINETD	?= y
+
+# SSH Packages :
+BR2_PACKAGE_OPENSSH		?= y
+BR2_PACKAGE_OPENSSH_SERVER	?= y
+BR2_PACKAGE_OPENSSH_KEY_UTILS	?= y
+
+# Openssl binary
+BR2_PACKAGE_LIBOPENSSL_BIN	?= y
+BR2_PACKAGE_LIBP11	?= y
+
+PLATFORM = AMD Versal Gen 2
+OPTEE_OS_PLATFORM = versal2
+OPTEE_OS_COMMON_EXTRA_FLAGS ?= CFG_PKCS11_TA=y CFG_USER_TA_TARGET_pkcs11=ta_arm64 O=out/arm
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+TF_A_PATH	?= $(ROOT)/arm-trusted-firmware
+U-BOOT_PATH	?= $(ROOT)/u-boot-xlnx
+LINUX_PATH	?= $(ROOT)/linux-xlnx
+
+include common.mk
+
+BINARIES_PATH	?= $(ROOT)/out/bin
+TF_A_ELF	?= $(TF_A_PATH)/build/$(OPTEE_OS_PLATFORM)/release/bl31/bl31.elf
+OPTEE_OS_ELF	?= $(OPTEE_OS_PATH)/out/arm/core/tee.elf
+U-BOOT_ELF	?= $(U-BOOT_PATH)/u-boot.elf
+U-BOOT_DTB	?= $(U-BOOT_PATH)/arch/arm/dts/versal2-*.dtb
+MKIMAGE_PATH	?= $(U-BOOT_PATH)/tools
+LINUX_IMAGE	?= $(LINUX_PATH)/arch/arm64/boot/Image
+ROOTFS_GZ	?= $(ROOT)/out-br/images/rootfs.cpio.gz
+ROOTFS_SIGN	?= $(BINARIES_PATH)/rootfs.cpio.gz.u-boot
+
+################################################################################
+# Targets
+################################################################################
+
+all: tfa optee-os dtbo u-boot linux buildroot buildroot_mkimg
+clean: tfa-clean optee-os-clean u-boot-clean linux-clean buildroot-clean dtbo-clean
```

Comment:
nitpicking: prefer the same dependency order in `all` and `clean` targets. Not a effective issue, just for consistency.

Reply:
Thanks @etienne-lms for the review. I have addressed the comments.